File: test-cache-demo/sample.php
<?php
// Sample PHP file with vulnerabilities
$userInput = $_GET['input'];

// Vulnerable: SQL Injection
$query = "SELECT * FROM users WHERE name = '" . $userInput . "'";
mysql_query($query);

// Vulnerable: XSS
echo "<div>" . $userInput . "</div>";

// Vulnerable: Command Injection
system("ls " . $_GET['path']);
?>

File: test-cache-demo/sample.go
package main

import (
    "fmt"
    "database/sql"
    _ "github.com/go-sql-driver/mysql"
)

func vulnerableQuery(userInput string) {
    db, _ := sql.Open("mysql", "user:password@/dbname")
    // Vulnerable: Direct string concatenation in SQL query
    query := "SELECT * FROM users WHERE name = '" + userInput + "'"
    rows, _ := db.Query(query)
    defer rows.Close()
}

func main() {
    fmt.Println("Demo application")
    vulnerableQuery("admin")
}
